src/TLuaInterpreter.cpp

```diff
@@ -9953,18 +9935,15 @@ int TLuaInterpreter::echoPopup(lua_State* L)
     }
 
     Host& host = getHostFromLua(L);
-    QString txt = a2.c_str();
-    QString name = a1.c_str();
-    if (_commandList.size() != _hintList.size()) {
-        lua_pushstring(L, "Error: command list size and hint list size do not match cannot create popup");
-        lua_error(L);
-        return 1;
+    if (commandList.size() != hintList.size()) {
+        lua_pushfstring(L, "echoPopup: commands and hints list aren't the same size");
+        return lua_error(L);
     }
 
-    if (a1.empty()) {
-        host.mpConsole->echoLink(txt, _commandList, _hintList, customFormat);
+    if (window.isEmpty()) {
+        host.mpConsole->echoLink(text, commandList, hintList, customFormat);
     } else {
-        mudlet::self()->echoLink(&host, name, txt, _commandList, _hintList, customFormat);
+        mudlet::self()->echoLink(&host, window, text, commandList, hintList, customFormat);
```

Comment:
Doesn't this one return a `bool` to indicated that the given `windowName` was found - does that not deserve returning to the user with `true`/`nil + error message` as appropriate?

Reply:
See above